arch/arm/src/gd32f4/hardware/gd32f4xx_spi.h

```diff
@@ -0,0 +1,222 @@
+
+#ifndef __ARCH_ARM_SRC_GD32F4_HARDWARE_GD32F4XX_SPI_H
+#define __ARCH_ARM_SRC_GD32F4_HARDWARE_GD32F4XX_SPI_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include "chip.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* SPIx(x=0,1,2,3,4,5) definitions */
+#define GD32_SPI0_BASE             (GD32_SPI_BASE+0x0000F800)      /* SPI0 base address */
+#define GD32_SPI1_BASE             (GD32_SPI_BASE+0x00000000)      /* SPI1 base address */
+#define GD32_SPI2_BASE             (GD32_SPI_BASE+0x00000400)      /* SPI2 base address */
+#define GD32_SPI3_BASE             (GD32_SPI_BASE+0x0000FC00)      /* SPI3 base address */
+#define GD32_SPI4_BASE             (GD32_SPI_BASE+0x00011800)      /* SPI4 base address */
+#define GD32_SPI5_BASE             (GD32_SPI_BASE+0x00011C00)      /* SPI5 base address */
+
+/* I2Sx_ADD(x=1,2) definitions */
+#define GD32_I2S1_ADD_BASE         (GD32_I2S_ADD_BASE+0x00000000)  /* I2S1_ADD base address */
+#define GD32_I2S2_ADD_BASE         (GD32_I2S_ADD_BASE+0x00000C00)  /* I2S2_ADD base address */
+
+/* Register Offsets *********************************************************/
+#define GD32_SPI_CTL0_OFFSET       0x0000      /* SPI control register 0 offset */
+#define GD32_SPI_CTL1_OFFSET       0x0004      /* SPI control register 1 offset */
+#define GD32_SPI_STAT_OFFSET       0x0008      /* SPI status register offset */
+#define GD32_SPI_DATA_OFFSET       0x000C      /* SPI data register offset */
+#define GD32_SPI_CRCPOLY_OFFSET    0x0010      /* SPI CRC polynomial register offset */
+#define GD32_SPI_RCRC_OFFSET       0x0014      /* SPI receive CRC register offset */
+#define GD32_SPI_TCRC_OFFSET       0x0018      /* SPI transmit CRC register offset */
+#define GD32_SPI_I2SCTL_OFFSET     0x001C      /* SPI I2S control register offset */
+#define GD32_SPI_I2SPSC_OFFSET     0x0020      /* SPI I2S clock prescaler register offset */
+#define GD32_SPI_QCTL_OFFSET       0x0080      /* SPI quad mode control register offset */
+
+/* Register Addresses *******************************************************/
+
+#define GD32_SPI0                  GD32_SPI0_BASE
+#define GD32_SPI1                  GD32_SPI1_BASE
+#define GD32_SPI2                  GD32_SPI2_BASE
+#define GD32_SPI3                  GD32_SPI3_BASE
+#define GD32_SPI4                  GD32_SPI4_BASE
+#define GD32_SPI5                  GD32_SPI5_BASE
+
+#define GD32_I2S1_ADD              GD32_I2S1_ADD_BASE
+#define GD32_I2S2_ADD              GD32_I2S2_ADD_BASE
+
+/* SPI registers definitions */
+#define GD32_SPI_CTL0(spix)        (spix+GD32_SPI_CTL0_OFFSET)      /* SPI control register 0 */
```

Comment:
```suggestion
#define GD32_SPI_CTL0(spix)        ((spix)+GD32_SPI_CTL0_OFFSET)      /* SPI control register 0 */
```
and other similar places